### PR TITLE
Aretomofix

### PR DESCRIPTION
**Pull request by Vilax**

### CODE REVIEW

**Comment by Vilax** [comment]
@azazellochg This PR should fix https://github.com/scipion-em/scipion-em-aretomo/issues/50

**Comment by azazellochg** [comment]
Does this mean the tilt angles will be always taken from input ts rather than ts associated with ctf?

**Comment by azazellochg** [comment]
What happens if I use imod to estimate ctf (which uses tilt geometry unlike ctffind) and also refine tilt angles in aretomo? If Imod does not raise an exception but only a warning maybe we should make sure that tilt angles match? Ctf correction will be wrong if we proceed as is.

**Comment by pconesa** [comment]
Do you mean imod ctf on a -60 degree image takes into account the tilt angle, and that this CTF estimation cannot, or is wrong to apply it to a -63 degrees angle (reestimated in a tilt series alignment)?

**Comment by Vilax** [comment]
@azazellochg I think I am missing something

CTF protocols use to estimate the defocus/astigmatism of the images. This is a mean value, and this value is stored in the CTF data model. Therefore, we have a CTF per image that is a mean value. This estimation does not take into account the tilt angle, does imod take it into account? or does imod carried out a local estimation or estimation per strip? Under these hypothesis, the fix should be correct

File: imod/protocols/protocol_xRaysEraser.py [hunk 1]
```diff
@@ -217,7 +217,9 @@ def createOutputStep(self, tsObjId):
         self._store()
 
     def closeOutputStep(self):
-        getattr(self, OUTPUT_TILTSERIES_NAME).setStreamState(Set.STREAM_CLOSED)
+        if self.TiltSeries:
+            self.TiltSeries.setStreamState(Set.STREAM_CLOSED)
+            self.TiltSeries.write()
```
[the remark this answers is not in the document]
Comment (pconesa) [reply imod/protocols/protocol_xRaysEraser.py:222]: I only see one write and 1 store

[the remark this answers is not in the document]
Comment (azazellochg) [reply imod/protocols/protocol_xRaysEraser.py:222]: output.write()

[the remark this answers is not in the document]
Comment (pconesa) [reply imod/protocols/protocol_xRaysEraser.py:222]: TiltSeries.write()

and self.store()

**Comment by Vilax** [comment]
> Good question. From my understanding imod does tile-based estimation on some kind of average over neighboring tilts. It might be that the exact tilt angle values are irrelevant but the overall tilt range/step is important. The ctf correction in imod is strip-based.
> 
> Anyway, I think I'm fine with this PR. If I'm not mistaken, FEI stage tilt accuracy is <0.1deg, so I never solve for tilt angles - they are more precise than any program can estimate from low contrast cryo-em data.

From my point of view, the imod tile based estimation carries out a tilt image average in order to get stronger Thon rings. But the average considers a tilt image and some tilt images around it according to the user choice. The idea is that the defocus does not vary very fast between consecutive tilt image. Even in this case I think the approach of the PR is correct.

**Pull request merged**